### ISSUE TITLE
Any kind of option to change caret's color

### ISSUE THREAD

**Issue by erkage** [vueform/multiselect#85]
Üdv! ;)

As i see know it's nearly impossible to change the caret's color, as the only way is to override the `.multiselect-caret`'s `background-image` css.

A possible solution is to change the `span` to the actual `svg` element and then the color of it can be changed with the css `fill` property.

Also this applies to other similar elements like the clear-icon.

**Comment by adamberecz** [comment]
The goal is to make the whole layout customizable via CSS. If we were using inline SVG that would break this goal and we were only able to change the icon by overriding it via slot.

However it's possible to change fill color when using SVG as a background pretty easily. Here's a blue caret:

``` css
background-image: url("data:image/svg+xml,%3Csvg viewBox='0 0 320 512'
fill='%230000FF' 
xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z'%3E%3C/path%3E%3C/svg%3E");
```

Mark the **fill='%230000FF'**. **%23** represents the **#**, which is followed by the hex color **0000FF**.

For Tailwind: I've just updated the readme and added a section I forgot before - the list of custom backgrounds that you need define: https://github.com/vueform/multiselect#styling-with-tailwind-css - here you can change the color as well.

**Comment by adamberecz** [comment]
Nevertheless, you might create your own version with custom `caret`, `clear` slots and implement inline SVGs.

**Comment by erkage** [comment]
Thank you for your answer.

I understand and agree with your css customization goal.
But as i mentioned in first comment i knew the css way to change it but i think it's just not elegant in that way and the inline svg was just my first and "hit on my stomach" idea to improve it.
Imho the Tailwind way is also bit "complex" => you need to install an extra package just for it.

What do you think about this alternative solution:
with: https://developer.mozilla.org/en-US/docs/Web/CSS/mask-image
you can do this:
```
.multiselect-caret {   
    mask-image: url("data:image/svg+xml;charset=utf-8,%3Csvg viewBox='0 0 320 512' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z'/%3E%3C/svg%3E");
    -webkit-mask-image: url("data:image/svg+xml;charset=utf-8,%3Csvg viewBox='0 0 320 512' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z'/%3E%3C/svg%3E");
    background-color: white;
}
```
and you can change the color with just one simple css line.
Only caveat there's no IE support but thank god it's obsolate now.

Again and of course this is just a suggestion, it's you decision and your way. I will able to handle it 🙂 .

**Comment by adamberecz** [comment]
Thanks for the elegant solution @erkage! It got implemented in `2.1.0`.

**Comment by jb-alvarado** [comment]
Hello, sorry that I write to a closed issue! 

Is this approach with mask-image also possible when using tailwind? Until now I have not find out how to solve this.
